MantidQt/CustomInterfaces/test/ImageROIPresenterTest.h

```diff
@@ -121,22 +143,64 @@ class ImageROIPresenterTest : public CxxTest::TestSuite {
     EXPECT_CALL(mockView, userError(testing::_, testing::_)).Times(0);
     EXPECT_CALL(mockView, userWarning(testing::_, testing::_)).Times(1);
 
+    // because the path is wrong this should not happen
+    EXPECT_CALL(mockView, resetWidgetsOnNewStack()).Times(0);
+
     // should not get there because there's no stack/img
     EXPECT_CALL(mockView, showStack(testing::An<const std::string &>()))
         .Times(0);
-    EXPECT_CALL(mockView,
-                showStack(testing::An<Mantid::API::WorkspaceGroup_sptr &>()))
+    EXPECT_CALL(
+        mockView,
+        showStack(testing::An<const Mantid::API::WorkspaceGroup_sptr &>(),
+                  testing::An<const Mantid::API::WorkspaceGroup_sptr &>(),
+                  testing::An<const Mantid::API::WorkspaceGroup_sptr &>()))
         .Times(0);
     EXPECT_CALL(mockView, updateImgWithIndex(testing::_)).Times(0);
 
     // this exception is currently handled, and a warning given
-    //TSM_ASSERT_THROWS("There should be an exception if there is an unexpected "
+    // TSM_ASSERT_THROWS("There should be an exception if there is an unexpected
+    // "
     //                  "error with the images path",
     //                  pres.notify(IImageROIPresenter::BrowseImgOrStack),
     //                  Poco::FileNotFoundException);
     pres.notify(IImageROIPresenter::BrowseImgOrStack);
   }
 
+  void test_changeImageType() {
+    testing::NiceMock<MockImageROIView> mockView;
+    MantidQt::CustomInterfaces::ImageROIPresenter pres(&mockView);
+
+    Mantid::API::WorkspaceGroup_sptr stack;
+    EXPECT_CALL(mockView, currentImageTypeStack())
+        .Times(1)
+        .WillOnce(Return(stack));
+    EXPECT_CALL(mockView, updateImageType(stack)).Times(1);
+
+    // should not mix up with the img index
+    EXPECT_CALL(mockView, currentImgIndex()).Times(0);
+    EXPECT_CALL(mockView, updateImgWithIndex(testing::_)).Times(0);
+
+    // Change without issues
+    EXPECT_CALL(mockView, userError(testing::_, testing::_)).Times(0);
+    EXPECT_CALL(mockView, userWarning(testing::_, testing::_)).Times(0);
+
+    pres.notify(IImageROIPresenter::ChangeImageType);
+  }
+
+  void test_changeRotation() {
+    testing::NiceMock<MockImageROIView> mockView;
+    MantidQt::CustomInterfaces::ImageROIPresenter pres(&mockView);
+
+    EXPECT_CALL(mockView, currentRotationAngle()).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(mockView, updateRotationAngle(0.0f)).Times(1);
+
+    // No errors, no warnings
+    EXPECT_CALL(mockView, userError(testing::_, testing::_)).Times(0);
+    EXPECT_CALL(mockView, userWarning(testing::_, testing::_)).Times(0);
+
+    pres.notify(IImageROIPresenter::ChangeRotation);
+  }
```

Comment:
In future, you might consider writing a final verification `TSM_ASSERT("Mock not used as expected", Mock::VerifyAndClearExpectations(&mockView))`, otherwise you get warnings from you `EXPECT_CALL`, but the test method will not register a failure for incorrect use.